network-mux/demo/cardano-ping.hs

```diff
@@ -160,13 +194,17 @@ main = do
     doLog :: StrictTMVar IO LogMsg -> LogMsg -> IO ()
     doLog msgQueue msg = atomically $ putTMVar msgQueue msg
 
-data NodeToNodeVersion = NodeToNodeVersionV1 Word32
-                       | NodeToNodeVersionV2 Word32
-                       | NodeToNodeVersionV3 Word32
-                       | NodeToNodeVersionV4 Word32 Bool
-                       | NodeToNodeVersionV5 Word32 Bool
-                       | NodeToNodeVersionV6 Word32 Bool
-                       | NodeToNodeVersionV7 Word32 Bool
+data NodeVersion =       NodeToClientVersionV5 Word32
+                       | NodeToClientVersionV6 Word32
+                       | NodeToClientVersionV7 Word32
+                       | NodeToClientVersionV8 Word32
+                       | NodeToNodeVersionV1   Word32
+                       | NodeToNodeVersionV2   Word32
+                       | NodeToNodeVersionV3   Word32
+                       | NodeToNodeVersionV4   Word32 Bool
+                       | NodeToNodeVersionV5   Word32 Bool
+                       | NodeToNodeVersionV6   Word32 Bool
+                       | NodeToNodeVersionV7   Word32 Bool
```

Comment:
It is a bit strange to mix both versions together.  Is this because `cardano-ping` accepts a single option for address which might be either a network address of local socket?

Reply:
There's a distinction between `node-to-node` and `node-to-client` versions with `supportedNodeToNodeVersions` and `supportedNodeToClientVersions` which introduces the split between `NodeVersion`s.  I would suggest to do that differently:

* make the `pingClient` polymorphic over versions it accepts
* have two different types for `node-to-client` and `node-to-node` versions; if there are any operations needed on versions they could be provided with a type class.

Reply:
That seems like an unnecessary abstraction.